src/com/palmergames/bukkit/util/BukkitTools.java

```diff
@@ -1,321 +1,323 @@
-	@Deprecated
-	public static byte getData(Block block) {
-		return block.getData();
-	}
-	// No Longer Used, used to be used in PlotBlockData's restorenextblock.
-	@Deprecated
-	public static void setTypeIdAndData(Block block, int type, byte data, boolean applyPhysics) {
-		Material mat = Material.getMaterial(IdMappings.getById(String.format("%s:%s", type, data)).getFlatteningType());
-		block.setType(mat, applyPhysics);		
-	}
-	// No Longer Used, used to be used in PlotBlockData's restorenextblock.
-	@Deprecated
-	public static void setTypeId(Block block, int type, boolean applyPhysics) {
-		Material mat = Material.getMaterial(IdMappings.getById(String.valueOf(type)).getFlatteningType());
-		block.setType(mat, applyPhysics);
-	}
-		
-	
-	/*
-	 * BlockState Methods
-	 */
-
-	public static Material getType(BlockState state) {
-		
-		return state.getType();
-	}
-	
-	public static MaterialData getData(BlockState state) {
-		
-		return state.getData();
-	}
-	
-	public static byte getDataData(BlockState state) {
-		
-		return getData(state).getData();
-	}
-	
-	
-	/*
-	 * Item Handling Methods
-	 */
-	
-	public static MaterialData getData(ItemStack stack) {
-		
-		return stack.getData();
-	}
-	
-	public static byte getDataData(ItemStack stack) {
-		
-		return getData(stack).getData();
-	}
-	
-	
-	/*
-	 * Material handling Methods.
-	 */
-	
-	/**
-	 * Find a Material from an Id.
-	 * Helpfully using Phoenix616's useful IdConverter.jar
-	 * https://www.spigotmc.org/resources/id-converter.52099/
-	 * 
-	 * @param id
-	 * @return
-	 */
-	public static Material getMaterial(int id) {
-		return Material.getMaterial(IdMappings.getById(String.valueOf(id)).getFlatteningType());
-	}
-	
-	/**
-	 * Find a Material from an enum name.
-	 * 
-	 * @param name
-	 * @return
-	 */
-	public static Material getMaterial(String name) {
-		
-		return Material.getMaterial(name);
-	}
-	
-	/**
-	 * Get the Id (magic number) of a Material type.
-	 * 
-	 * @param material
-	 * @return
-	 */
-	@Deprecated
-	public static int getMaterialId(Material material) {
-		
-		return material.getId();
-	}
-	
-
-	/**
-	 * Compiles a list of all whitelisted users.
-	 * 
-	 * @return List of all whitelist player names.
-	 */
-	public static List<String> getWhiteListedUsers() {
-
-		List<String> names = new ArrayList<String>();
-		try {
-			BufferedReader fin = new BufferedReader(new FileReader("white-list.txt"));
-
-			try {
-				String line;
-				while ((line = fin.readLine()) != null)
-					names.add(line);
-			} catch (IOException e) {
-			}
-
-			try {
-				fin.close();
-			} catch (IOException e) {
-			}
-		} catch (FileNotFoundException e) {
-		}
-		return names;
-	}
-
-	/**
-	 * Accepts an X or Z value and returns the associated Towny plot value.
-	 * 
-	 * @param value
-	 * @return int of the relevant townblock x/z.
-	 */
-	public static int calcChunk(int value) {
-
-		return (value * TownySettings.getTownBlockSize()) / 16;
-	}
-}
+package com.palmergames.bukkit.util;
```

Comment:
- Fix plot perm line not being set properly on plots owned by someone who leaves town.